### PR TITLE
Fix parent index of repeated compoonents

### PR DESCRIPTION
**Pull request by hunger**
Done with @tronical

### CODE REVIEW

**Comment by tronical** [comment]
What about the interpreter?

File: internal/compiler/generator/rust.rs [hunk 1]
```diff
@@ -1023,12 +1023,22 @@ fn generate_item_tree(
...
+                .upgrade()
+                .map(|sc| (VRcMapped::origin(&sc), sc.tree_index_of_first_child.get()))
+            {
+                *result = slint::re_exports::ItemRc::new(parent_component, parent_index as usize + #sub_component_offset - 1)
+                    .downgrade();
+            })
+        })
     });
-    let parent_item_index = parent_item_index.iter();
+    let parent_item_expression = parent_item_expression.iter();
```
Comment (ogoffart) [inline internal/compiler/generator/rust.rs:1041]:
```suggestion
```

This doesn't need to be an iterator  if you don't add more to the expention in the quote! macro.  (quote expends None to nothing)

(Also need to remove the `#(` and `)*`  of course)

**Comment by ogoffart** [comment]
> What about the interpreter?

I think this isn't an issue in the interpreter because it is all inlined

File: internal/compiler/generator/rust.rs (hunk 1 above)
Comment (hunger) [reply internal/compiler/generator/rust.rs]: Good point, fixed!

**Pull request merged**